File: Java/src/com/leetcode/Easy/KthDistinctStringInAnArray.java
package com.leetcode.Easy;

import java.util.*;

// 2053. Kth Distinct String in an Array
public class KthDistinctStringInAnArray {
    public String kthDistinct(String[] arr, int k) {
        Map<String, Integer> map = new LinkedHashMap<>();
        for (String str : arr) {
            map.put(str, map.getOrDefault(str, 0) + 1);
        }

        for (Map.Entry<String, Integer> entry : map.entrySet()) {
            if (entry.getValue() == 1) {
                k--;
                if (k == 0) return entry.getKey();
            }
        }

        return "";
    }
}

File: JavaScript/Easy/KthDistinctStringInAnArray.js
/**
 * 2053. Kth Distinct String in an Array
 * @param {string[]} arr
 * @param {number} k
 * @return {string}
 */
const kthDistinct = function (arr, k) {
  let map = new Map();
  for (let a of arr) {
    map.set(a, (map.get(a) || 0) + 1);
  }

  const distinctStrings = [];
  for (const [str, count] of map.entries()) {
    if (count === 1) {
      distinctStrings.push(str);
    }
  }

  return distinctStrings.length >= k ? distinctStrings[k - 1] : "";
};